src/shell/builtin/yes.zig

```diff
@@ -1,37 +1,115 @@
-state: enum { idle, waiting_io, err, done } = .idle,
+state: enum { idle, waiting_write_err, waiting_io, err, done } = .idle,
 expletive: []const u8 = "y",
 task: YesTask = undefined,
+buffer: []u8 = "",
+buffer_used: usize = 0,
+alloc_scope: shell.AllocScope,
 
 pub fn start(this: *@This()) Yield {
     const args = this.bltn().argsSlice();
 
-    if (args.len > 0) {
-        this.expletive = std.mem.sliceTo(args[0], 0);
+    // count
+    var bufalloc: usize = 0;
+    if (args.len == 0) {
+        bufalloc = 2; // "y\n"
+    } else {
+        // Sum all args + spaces between + newline
+        for (args, 0..) |arg, i| {
+            const arg_slice = std.mem.sliceTo(arg, 0);
+            bufalloc += arg_slice.len;
+            if (i < args.len - 1) bufalloc += 1; // space
+        }
+        bufalloc += 1; // newline
+    }
+
+    // Use at least BUFSIZ (8192) for better performance
+    const BUFSIZ = 8192;
+    if (bufalloc <= BUFSIZ / 2) {
+        bufalloc = BUFSIZ;
+    }
+
+    this.buffer = this.alloc_scope.allocator().alloc(u8, bufalloc) catch bun.outOfMemory();
+
+    // Fill buffer with one copy of the output
+    this.buffer_used = 0;
+    if (args.len == 0) {
+        @memcpy(this.buffer[0..1], "y");
+        this.buffer[1] = '\n';
+        this.buffer_used = 2;
+    } else {
+        for (args, 0..) |arg, i| {
+            const arg_slice = std.mem.sliceTo(arg, 0);
+            @memcpy(this.buffer[this.buffer_used .. this.buffer_used + arg_slice.len], arg_slice);
+            this.buffer_used += arg_slice.len;
+            if (i < args.len - 1) {
+                this.buffer[this.buffer_used] = ' ';
+                this.buffer_used += 1;
+            }
+        }
+        this.buffer[this.buffer_used] = '\n';
+        this.buffer_used += 1;
     }
 
+    // Fill larger buffer by repeating the pattern
+    const copysize = this.buffer_used;
+    var copies = bufalloc / copysize;
+    var filled = this.buffer_used;
+    while (copies > 1) : (copies -= 1) {
+        const remaining = bufalloc - filled;
+        const to_copy = @min(copysize, remaining);
+        @memcpy(this.buffer[filled .. filled + to_copy], this.buffer[0..to_copy]);
+        filled += to_copy;
+    }
+    this.buffer_used = filled;
+
     if (this.bltn().stdout.needsIO()) |safeguard| {
         const evtloop = this.bltn().eventLoop();
         this.task = .{
             .evtloop = evtloop,
             .concurrent_task = JSC.EventLoopTask.fromEventLoop(evtloop),
         };
         this.state = .waiting_io;
-        this.task.enqueue();
-        return this.bltn().stdout.enqueueFmt(this, "{s}\n", .{this.expletive}, safeguard);
+        return this.bltn().stdout.enqueue(this, this.buffer[0..this.buffer_used], safeguard);
```

Comment:
This amount of complexity for something that should be simple is likely to create new crashes.

If you want to implement a fill repeating function it should be moved out of `src/shell/builtin/yes.zig` (and perhaps the existing one in `Buffer.zig` could be more generalized)